chapter_recurrent-neural-networks/encoder-decoder.md

```diff
@@ -27,7 +26,7 @@ class Encoder(nn.Block):
 
 ## Decoder
 
-The decoder has an additional method `init_state` to parse the outputs of the encoder with possible additional information, e.g., the valid lengths of inputs, to return the state it needs. In the forward method, the decoder takes both inputs, e.g., a target sentence, and the state. It returns outputs, with potentially modified state if the encoder contains RNN layers.
+The decoder has an additional method `init_state` to parse the outputs of the encoder (the states) with possible additional information, e.g., returns the state it needs given the valid lengths of inputs. In the forward method, the decoder takes both inputs, e.g., a target sentence and the state. It returns outputs, with potentially modified state if the encoder contains RNN layers.
```

Comment:
possible additional information, e.g., the valid lengths of inputs

Reply:
sure